app/pages/project/home/home-workflow-button.spec.js

```diff
@@ -55,9 +77,29 @@ describe('ProjectHomeWorkflowButton', function() {
   it('calls handleWorkflowSelection onClick', function() {
     wrapper.find('Link').simulate('click');
     assert.equal(handleWorkflowSelectionSpy.calledOnce, true);
-    assert.equal(onChangePreferencesSpy.calledOnce, true);
   });
 
+  it('should update user preferences on workflow selection', function (done) {
+    wrapper.instance().handleWorkflowSelection()
+    .then(function () {
+      assert.equal(preferences.update.calledOnce, true);
+    })
+    .then(done, done);
+  });
+  it('should clear the current workflow', function (done) {
+    wrapper.instance().handleWorkflowSelection()
+    .then(function () {
+      assert.equal(actions.classifier.setWorkflow.calledWith(null), true);
+    })
+    .then(done, done);
+  });
+  it('should select a new workflow', function (done) {
```

Comment:
This test, and the preceding test, should really test that the old workflow is cleared before the new workflow is set. I wasn't quite sure how to do that with sinon.

Reply:
Fixed now, thanks to @rogerhutchings advice.